src/upgrades/upgradeable.cairo

```diff
@@ -11,15 +17,22 @@ mod Upgradeable {
         Upgraded: Upgraded
     }
 
+    /// Emitted when the contract is upgraded.
     #[derive(Drop, starknet::Event)]
     struct Upgraded {
         class_hash: ClassHash
     }
 
+    mod Errors {
+        const INVALID_CLASS: felt252 = 'Class hash cannot be zero';
+    }
+
     #[generate_trait]
-    impl InternalImpl of InternalState {
-        fn _upgrade(ref self: ContractState, new_class_hash: ClassHash) {
-            assert(!new_class_hash.is_zero(), 'Class hash cannot be zero');
+    impl InternalImpl<
+        TContractState, +HasComponent<TContractState>
+    > of InternalTrait<TContractState> {
+        fn _upgrade(ref self: ComponentState<TContractState>, new_class_hash: ClassHash) {
```

Comment:
do we still need the underscore?

Reply:
https://github.com/OpenZeppelin/cairo-contracts/pull/794#discussion_r1366078416

Andrew made a good point, we should define if we need it or not first, right?

Reply:
let's keep it for now and have that discussion then, did we have an issue for this?

Reply:
https://github.com/OpenZeppelin/cairo-contracts/issues/795